_pages/dev/ssr/server-side-rendering-contextual-logging.md

```diff
@@ -0,0 +1,244 @@
+```
+{"message":"Rendering started (/powertools-spa/en/USD/product/3788616/bt-dy-720-e)","context":{"timestamp":"2023-06-19T23:34:37.845Z","request":{"url":"/powertools-spa/en/USD/product/3788616/bt-dy-720-e","uuid":"1ef6d1d9-65c2-43c8-860e-32d0cb1bf698","timeReceived":"2023-06-19T23:34:37.843Z"}}}
+```
+The following is the representation of the log message in the monitoring tool:
+![Log Representation In Kibana](assets/images/../../../../../assets/images/ssr/contextual_logging_kibana_logs.png)
+
+
+
+## Enabling the SSR Contextual Logging
+
+To enable the SSR contextual logging, add the `logger` flag to the configuration object as the second parameter to the method that decorates `ngExpressEngine` in the `server.ts` file. The following is an example:
+
+```ts
+import { ngExpressEngine as engine } from '@nguniversal/express-engine';
+import { NgExpressEngineDecorator } from '@spartacus/setup/ssr';
+
+[...]
+
+const ngExpressEngine = NgExpressEngineDecorator.get(engine, {
+  //add logger flag here
+  logger: true
+});
+
+[...]
+
+```
+
+From now on, your application will use the `DefaultExpressServerLogger` and provide logging context for all server-side rendered requests
+
+## Customizing the SSR Contextual Logging
+
+The Composable Storefront offers developers the ability to customize the SSR Contextual Logging. This includes the option to provide a custom logger in place of the default one. 
+
+To implement a custom logger, the developer must create a class that implements the `ExpressServerLogger` interface. Alternatively, they may extend the `DefaultExpressServerLogger` class. 
+
+Once the custom logger class is created, it can be passed to the `logger` property in the configuration object. 
+
+An example of how to do this is shown below:
+
+```ts
+import { ngExpressEngine as engine } from '@nguniversal/express-engine';
+import { NgExpressEngineDecorator } from '@spartacus/setup/ssr';
+import { CustomExpressServerLogger } from './custom-express-server-logger';
+
+[...]
+
+const ngExpressEngine = NgExpressEngineDecorator.get(engine, {
+  // pass custom logger here
+  logger: new CustomExpressServerLogger()
+});
+
+[...]
+```
+
+The following is an example of a custom logger implementation:
+
+```ts
+import { ExpressServerLogger } from '@spartacus/setup/ssr';
+
+export class CustomExpressServerLogger implements ExpressServerLogger {
+
+  //custom implementation of the `log` method
+  log(message: string, context: ExpressServerLoggerContext): void {
+    console.log({ message: `[My Custom LOG]: ${message}`, ...context });
+  }
+
+  [...]
+}
+```
+
+An important aspect of our solution is its extensibility. We have designed our solution to be compatible with enterprise logging libraries used for logging in platform-server such as [Winston](https://www.npmjs.com/package/winston) or [Pino](https://www.npmjs.com/package/pino), making it easy to add their features to our system.
```

Comment:
Let's divide the following section into 2 subsections with 2 headers:
- Plugging third party logger
- Providing additional context to logs

so the separate the examples of the customization usecases